src/java_bytecode/jar_file.cpp

```diff
@@ -96,16 +138,17 @@ std::string jar_filet::get_entry(std::size_t i)
 
   std::string dest;
 
+  size_t real_index=filtered_jar[i];
```

Comment:
Apologies for having been unclear: I was referring to `i`, thus `assert(i<filtered_jar.size())` would cover that.